### PR TITLE
[feature] [CUDA solver] Add multi-GPU and ask for CUDA during btcli run

### LINKED ISSUE #897: Incapable of running `btcli run --cuda` on WSL ubuntu
**Describe the bug**
I have followed the [bittensor installation guide](https://docs.bittensor.com/RunningAMiner.html) verbatim on a Windows machine running WSL. 

Everything was running perfectly, returning "working" results, up until I was asked to run `btcli run --cuda` to run bittensor on my GPU. 

The initial errors I received had to do with `cubit` not being  found as a dependency (patebin [here](https://pastebin.com/wCx35hx3)), so I ended up installing it from source in the /.bittensor directory and pip installing it accordingly. I made some progress, but after running the `btcli` command with the cuda flag, I continued to receive stack traces telling me that a `cubit.pyx` file was not being found in my home directory. After copying the same file from the cubit repository I cloned earlier to my home directory, I've hit a wall and a stack trace I can't seem to route around.

This is where I've landed:

https://pastebin.com/BXwsQxs6

**To Reproduce**
1. Follow the [installation guide ](https://docs.bittensor.com/RunningAMiner.html) verbatim
2. Run into errors after running `btcli run --cuda`

**Screenshots**
My first stack trace, saying cubit wasn't installed, even when I followed the installation steps successfully:

My second stack trace, after installing cubit from source:

My third, intractable stack trace, saying I need an integer:

**Environment:**
 Windows 11 running WSL2:
 ```
 DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
```
 - Bittensor Version [e.g. 22]
 the latest release

**Additional context**

### PR DESCRIPTION
**Pull request by camfairchild**
This PR adds  
- Multi-GPU registration capability (one process per GPU + master)
    - `btcli register --cuda.dev_id 0 1 2 3 --cuda.use_cuda`
- A prompt for CUDA registration during `btcli run`
    - can skip this prompt with `btcli run --wallet.reregister false`

### CODE REVIEW

File: bittensor/_cli/__init__.py
Comment (unconst) [inline bittensor/_cli/__init__.py:843]: Nice you added the all command

**Review by unconst — APPROVED** [review]

**Comment by unconst** [comment]
will this also solve the 
btcli register --help 
not showing available arguments?

**Comment by camfairchild** [comment]
> will this also solve the 
> btcli register --help 
> not showing available arguments?

as long as the args are added under the register command parser (which they are) it'll show in the help usage

**Comment by unconst** [comment]
What is the status of this PR?

**Comment by camfairchild** [comment]
> What is the status of this PR?

waiting for the CI to pass again before merge into nobuanga.
We should merge master into nobuanga again first

**Pull request merged**
